homeassistant/components/matter/water_valve.py

```diff
@@ -0,0 +1,224 @@
+        )
+
+    async def async_set_valve_position(self, position: int) -> None:
+        """Move the water valve to a specific position."""
+        position = kwargs[ATTR_POSITION]
+        await self.send_device_command(
+            # A value of 100 percent SHALL indicate the fully open position
+            # A value of 0 percent SHALL indicate the fully closed position
+            # A value of null SHALL indicate that the current state is not known
+            clusters.ValveConfigurationAndControl.Commands.Open(
+                position
+            )  # TargetLevel type="percent"
+        )
+
+    async def send_device_command(self, command: Any) -> None:
+        """Send device command."""
+        await self.matter_client.send_device_command(
+            node_id=self._endpoint.node.node_id,
+            endpoint_id=self._endpoint.endpoint_id,
+            command=command,
+        )
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        operational_status = self.get_matter_attribute_value(
+            clusters.ValveConfigurationAndControl.Attributes.OperationalStatus
+        )
+
+        assert operational_status is not None
+
+        LOGGER.debug(
+            "Operational status %s for %s",
+            f"{operational_status:#010b}",
+            self.entity_id,
+        )
+
+        state = operational_status & OPERATIONAL_STATUS_MASK
+        match state:
+            # Valve is transitioning between closed and open positions or between levels
+            case OperationalStatus.VALVE_IS_CURRENTLY_TRANSITIONING:
+                self._attr_is_opening = True
+                # self._attr_is_closing = False
+                # self._attr_is_transitioning = True --> Not implemented here: https://developers.home-assistant.io/docs/core/entity/valve
+            case _:
+                self._attr_is_opening = False
+                self._attr_is_closing = False
+
+        if self._entity_info.endpoint.has_attribute(
+            None, clusters.ValveConfigurationAndControl.Attributes.CurrentLevel
+        ):
+            # A value of 100 percent SHALL indicate the fully open position
+            # A value of 0 percent SHALL indicate the fully closed position
+            # A value of null SHALL indicate that the current state is not known
+            current_valve_position = self.get_matter_attribute_value(
+                clusters.Attributes.CurrentLevel
+            )
+            self._attr_current_valve_position = (
+                floor(current_valve_position / 100)
+                if current_valve_position is not None
+                else None
+            )
+
+            LOGGER.debug(
+                "Current position for %s - raw: %s - corrected: %s",
+                self.entity_id,
+                current_valve_position,
+                self.current_valve_position,
+            )
+
+        self._attr_device_class = ValveDeviceClass.WATER
```

Comment:
no need to set this in each atrribute callback, move it to the ValveEntityDescription